archiver.go

```diff
@@ -1280,6 +1281,121 @@ func DeleteArchivedMessages(ctx context.Context, config *Config, db *sqlx.DB, s3
 	return nil
 }
 
+// DeleteBroadcasts deletes all broadcasts older than 90 days for the passed in org which have no active messages on them
+func DeleteBroadcasts(ctx context.Context, now time.Time, config *Config, db *sqlx.DB, org Org) error {
+	start := time.Now()
+	threshhold := now.AddDate(0, 0, -org.ActiveDays)
+
+	rows, err := db.QueryxContext(ctx, selectOldOrgBroadcasts, org.ID, threshhold)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	count := 0
+	for rows.Next() {
+		if count == 0 {
+			logrus.WithField("org_id", org.ID).Info("deleting broadcasts")
+		}
+
+		// been deleting this org more than an hour? thats enough for today, exit out
+		if time.Since(start) > time.Hour {
+			break
+		}
+
+		var broadcastID int64
+		err := rows.Scan(&broadcastID)
+		if err != nil {
+			return errors.Wrap(err, "unable to get broadcast id")
+		}
+
+		// make sure we have no active messages
+		var msgCount int64
+		err = db.Get(&msgCount, `SELECT count(*) FROM msgs_msg WHERE broadcast_id = $1`, broadcastID)
```

Comment:
maybe not so important but if you just want to know if any rows exist, don't need to full count

Reply:
Ya, felt the error message was more interesting with count. In theory we should never really get to this case anyways.